tools/protodoc/protodoc.py

```diff
@@ -514,23 +526,35 @@ def FormatFieldAsDefinitionListItem(outer_type_context, type_context, field):
   Returns:
     RST formatted definition list item.
   """
+  annotations = []
   if field.HasField('oneof_index'):
-    oneof_template = '\nPrecisely one of %s must be set.\n' if type_context.oneof_required[
-        field.oneof_index] else '\nOnly one of %s may be set.\n'
-    oneof_comment = oneof_template % ', '.join(
-        FormatInternalLink(
-            f, FieldCrossRefLabel(outer_type_context.ExtendField(0, f).name))
-        for f in type_context.oneof_fields[field.oneof_index])
+    oneof_context = outer_type_context.ExtendOneof(field.oneof_index,
+                                                   type_context.oneof_names[field.oneof_index])
+    oneof_comment, oneof_comment_annotations = oneof_context.LeadingCommentPathLookup()
+    if NOT_IMPLEMENTED_HIDE_ANNOTATION in oneof_comment_annotations:
+      return ''
+
+    if len(type_context.oneof_fields[field.oneof_index]) == 1 and type_context.oneof_required[
+        field.oneof_index]:
+      annotations = ['*REQUIRED*']
+
+    if len(type_context.oneof_fields[field.oneof_index]) > 1:
+      oneof_template = '\nPrecisely one of %s must be set.\n' if type_context.oneof_required[
+          field.oneof_index] else '\nOnly one of %s may be set.\n'
+      oneof_comment += oneof_template % ', '.join(
+          FormatInternalLink(
+              f, FieldCrossRefLabel(outer_type_context.ExtendField(i, f).name))
+          for i, f in type_context.oneof_fields[field.oneof_index])
   else:
     oneof_comment = ''
+
   anchor = FormatAnchor(FieldCrossRefLabel(type_context.name))
-  annotations = []
   if field.options.HasExtension(validate_pb2.rules):
     rule = field.options.Extensions[validate_pb2.rules]
     if ((rule.HasField('message') and rule.message.required) or
         (rule.HasField('string') and rule.string.min_bytes > 0) or
         (rule.HasField('repeated') and rule.repeated.min_items > 0)):
-      annotations.append('*REQUIRED*')
+      annotations = ['*REQUIRED*']
```

Comment:
Why is this safe, i.e. is this case mutually exclusive with the `oneof`? Looking at this without paging everything back in makes my head hurt (I know, it's my code originally). Can you add some comments?

Reply:
This is safe because annotations is not used for anything else than `*REQUIRED*`, it is not mutually exclusive with `oneof` so I changed from append to = to avoid have two `*REQUIRED*` in the list.

Reply:
Updated this to be more clear, and make `REQUIRED` less confusing. PTAL